### ISSUE TITLE
label embedding

### ISSUE THREAD

**Issue by Turlan** [tarun005/USSS_ICCV19#2]
Thanks for releasing the codes. Could you also release the implementation of label embedding?

**Comment by tarun005** [comment]
The label embeddings are computed using pretraining the same network with the limited labeled data. We will try to release the dataset specific embeddings soon.

**Comment by Turlan** [comment]
Thanks

**Comment by ShoufaChen** [comment]
Hi, do you have a plan to release the dataset specific embeddings soon?

**Comment by tarun005** [comment]
Hi Shoufa,

We will try to release the embedding soon, but they can also be computed using the given code.

Please follow section B.1 of the supplementary section of the paper, where we detailed the algorithm used to compute the embeddings.

In fact, we also note that for small N, you can use the word2vec representations of the labels as embedding to get comparable (or, better) performance.

**Comment by philadias** [comment]
Hi, could you please further detail how you performed the dimensionality reduction? My understanding is that one needs to:
1) using a model pretrained on the dataset, get the encoder outputs F(.) for all samples in the training set 
2) upsample them to original input resolution
3) using all these stored maps, finally perform PCA. 

I'm sure I'm missing some simplification, since this would require a large amount of memory. Did you maybe restrict to only a few samples? 

Or performed the PCA on the downsampled maps, and later upsampled only the dim.-reduced maps to compare with label maps for centroid computation? Thanks!

**Comment by tarun005** [comment]
@philadias You are right that for semantic segmentation, dimensionality reduction using each pixel in each image is computationally expensive. So, we subsampled some pixels from each feature map (and corresponding label map) from each image (making sure each class is represented well enough ) and then applied PCA on this _much_ reduced set. There might be other faster/better alternatives which we did not explore.
